File: src/main/java/cymru/asheiou/ashutils/command/admin/ashutils/ClearChatHandler.java
package cymru.asheiou.ashutils.command.admin.ashutils;

import cymru.asheiou.ashutils.sender.MessageSender;
import org.bukkit.entity.Player;
import org.bukkit.plugin.java.JavaPlugin;

public class ClearChatHandler {
  public static boolean doClearChat(JavaPlugin plugin) {
    for (Player player : plugin.getServer().getOnlinePlayers()) {
      if (!player.hasPermission("ashutils.clearchat.exempt")) {
        for (int i = 0; i < 100; i++) {
          player.sendMessage(" ");
        }
      } else {
        MessageSender.sendMessage(player, "Chat cleared for non-exempt users.");
      }
    }
    return true;
  }
}

File: src/main/kotlin/xyz/aeolia/lib/command/admin/util/ClearChatHandler.kt
package xyz.aeolia.lib.command.admin.util

import net.kyori.adventure.text.Component
import org.bukkit.Bukkit
import xyz.aeolia.lib.sender.MessageSender

class ClearChatHandler {
  companion object {
    @JvmStatic
    fun doClearChat() : Boolean {
      for (player in Bukkit.getOnlinePlayers()) {
        if(!player.hasPermission("libls.clearchat.exempt")) {
          repeat (100) {
            player.sendMessage(Component.text(" "))
          }
        } else {
          MessageSender.sendMessage(player, "Chat cleared for non-exempt users.")
        }
      }
      return true
    }
  }
}